pkg/api/types.go

```diff
@@ -2070,10 +2070,83 @@ const (
 	StrategicMergePatchType PatchType = "application/strategic-merge-patch+json"
 )
 
+// Type and constants for component type.
+type ComponentType string
+
+// ComponentType constants define the known types of components.
+// Each type may have multiple instances.
+// Each instance's name is prefixed by the type, with a random suffix. See SimpleNameGenerator.
```

Comment:
After digging around, I can't find where Pod creation populates GenerateName. So I guess I'll just put it in PrepareForCreate if Name is empty. I assume the etcd create will fail if a key with the same name already exists.